packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -126,6 +140,61 @@ export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedO
         this.attachListeners();
     }
 
+    /**
+     * Marks this objects as closed. Any attempt to change it (local changes or processing remote ops)
+     * would result in same error thrown. If called multiple times, only first error is remembered.
+     * @param error - error object that is thrown whenever an attempt is made to modify this object
+     */
+    protected closeWithError(error: any) {
+        if (this.closeError === undefined) {
+            this.closeError = error;
+        }
+    }
+
+    /**
+     * Verifies that this object is not closed via closeWithError(). If it is, throws an error used to close it.
+     */
+     protected verifyNotClosed() {
+        if (this.closeError !== undefined) {
+            throw this.closeError;
+        }
+    }
+
+    /**
+     * Event listener handler helper that can be used to react to exceptions thrown from event listeners
+     * It wraps error with DataProcessingError, closes this object and throws resulting error.
+     * See closeWithError() for more details
+     * Ideally such situation never happens, as consumers of DDS should never throw exceptions
+     * in event listeners (i.e. catch any of the issues and make determination on how to handle it).
+     * When such exceptions propagate through, most likely data model is no longer consistent, i.e.
+     * DDS state does not match what user sees. Because of it DDS moves to "corrupted state" and does not
+     * allow processing of ops or local changes, which very quickly results in container closure.
+     */
+     protected eventListenerErrorHandler(event: string | symbol, e: any) {
+        const error = DataProcessingError.wrapIfUnrecognized(
+            e,
+            "SharedObjectEventListenerException",
+            undefined); // message
+        error.addTelemetryProperties({ emittedEventName: String(event) });
+
+        this.closeWithError(error);
+        throw error;
+    }
+
+    /**
+     * Helper method that can be used to emit events and respond to exceptions thrown from event listeners
+     * by closing this object. Please see eventListenerErrorHandler() & closeWithError() for more info.
+     * @param event - event to raise
+     * @param args - payload of event
+     */
+    protected safeRaiseEvent(event, ...args) {
```

Comment:
Probably default to making all of these private until we have motivation for protected, easier to open up than to close down the interface.